### PR TITLE
Restructured code

### PR DESCRIPTION
**Pull request by janosdebugs**
## Please describe the change you are making

Restructuring code so that each entity has its own scope, together with mock and live client. Reasons:

1. Easier to spot duplicate code.
2. Mocks are treated as first-class citizens and are harder to forget.
3. Easier code navigation.

## Are you the owner of the code you are sending in, or do you have permission of the owner?

Yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

Yes

### CODE REVIEW

File: disk_list.go [hunk 1]
```diff
@@ -29,3 +29,15 @@ func (o *oVirtClient) ListDisks(retries ...RetryStrategy) (result []Disk, err er
 		})
 	return
 }
+
+func (m *mockClient) ListDisks(_ ...RetryStrategy) ([]Disk, error) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	result := make([]Disk, len(m.disks))
+	i := 0
```
Comment (engelmi) [inline disk_list.go:37]: Not really of this PR, but could we move the iterator variable `i` into the `for` loop?

**Review by eslutsky — COMMENTED** [review]
besides that the live tests failing with :
    disk_downloadimage_test.go:42: failed to upload image (bad_argument: Disk size must be at least 1048576 bytes (1 MB))
--- FAIL: TestImageDownload (1.10s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x893e65]

ooks good to me

**Comment by janosdebugs** [comment]
@michalskrivanek @emesika the failure above is reproducible, the OST tests should have caught it. Something is wrong with the result reporting.

**Comment by janosdebugs** [comment]
Merging, bugfixing the OST bug later.

**Pull request merged**